source/host/Option.cc

```diff
@@ -0,0 +1,52 @@
+#include "Option.hh"
+
+#include <cstring>
+
+namespace Host::Option {
+
+std::optional<EOption> CheckFlag(const char *arg) {
+    ASSERT(arg);
+    if (arg[0] != '-') {
+        return std::nullopt;
+    }
+
+    // Verbose flag
+    if (arg[1] == '-') {
+        const char *verbose_arg = &arg[2];
+
+        if (strcmp(verbose_arg, "mode") == 0) {
+            return EOption::Mode;
+        }
+
+        if (strcmp(verbose_arg, "suite") == 0) {
+            return EOption::Suite;
```

Comment:
Not sure I like the implications of having `EOption::Suite`. This option is specific to `KTestSystem` right? So why do we need to expose this here? What happens if we make a new mode where it makes sense to name a flag `-s`? Well `-s` is already taken for `EOption::Suite`, so any new mode flags must only use the remaining alphabet.

Reply:
Yes, I had considered this as well. I think the implementation is good despite these points. 

> This option is specific to `KTestSystem` right? So why do we need to expose this here?

I don't think so. This flag can be passed into the CLI regardless of mode, but the mode can decide how it chooses to interpret it.

> What happens if we make a new mode where it makes sense to name a flag `-s`? Well `-s` is already taken for `EOption::Suite`, so any new mode flags must only use the remaining alphabet.

We can cross this bridge when we get to it, but ultimately we can manage this however we want. An important note is that we can always have some flag along the lines of `-sb`, for example. We're not limited to single letters - we can nest logic in the switch statement cases however we choose to.

Reply:
Alright valid points. I'm okay with this then